draft-ietf-dnsop-svcb-https.md

```diff
@@ -495,59 +495,53 @@ is the effective TargetName:
-   connection to this endpoint fails, the client SHOULD try to connect using
-   values from the next-highest-priority compatible record, etc. If all attempts
-   fail, clients SHOULD go to step 5 (except as noted in {{ech-client-behavior}}).
+4. If one or more "compatible" ({{mandatory}}) ServiceMode records are returned,
+   these represent the alternative endpoints.
 
-5. At this point there are no usable ServiceMode records, because
-   - there were no SVCB records found for $SVCB_QNAME, OR
-   - all records found were incompatible with this client, OR
-   - all connection attempts using ServiceMode records failed.
-
-   Accordingly, clients SHALL connect to the
-   endpoint consisting of $ADDR_QNAME, the authority endpoint's
-   port number, and no SvcParams.
-
-If all of the above connection attempts fail, clients MAY connect to the
-authority endpoint (except as noted in {{client-failures}} and
-{{ech-client-behavior}}).
+5. Otherwise, SVCB resolution has failed.
 
 This procedure does not rely on any recursive or authoritative DNS server to
 comply with this specification or have any awareness of SVCB.
 
-When selecting between AAAA and A records to use, clients may use an approach
-such as Happy Eyeballs {{!HappyEyeballsV2=RFC8305}}.
+Once SVCB resolution is completed, the client proceeds with connection
+establishment.  Clients SHOULD try higher-priority alternatives first, with
+fallback to lower-priority alternatives.  Clients MUST issue AAAA and/or A
+queries for the selected TargetName, and MAY choose between them using an
+approach such as Happy Eyeballs {{!HappyEyeballsV2=RFC8305}}.
+
+A client is called "SVCB-reliant" if it connects only to the indicated
+ServiceMode endpoints, and "SVCB-optional" otherwise.  Clients for pre-existing
+protocols SHALL implement SVCB-optional behavior (except as noted in
+{{client-failures}} and {{ech-client-behavior}}).
+
+SVCB-optional clients SHOULD issue in parallel any other DNS queries that might
+be needed for connection establishment.  SVCB-optional clients SHALL append an
+alternative endpoint consisting of the final value of $QNAME, the authority
+endpoint's port number, and no SvcParams, and MAY append the authority endpoint.
 
 Some important optimizations are discussed in {{optimizations}}
 to avoid additional latency in comparison to ordinary AAAA/A lookups.
 
 ## Handling resolution failures {#client-failures}
 
-If a SVCB query results in a SERVFAIL error, transport error, or timeout,
+If SVCB resolution fails due to a SERVFAIL error, transport error, or timeout,
 and DNS exchanges between the client and the recursive resolver are
 cryptographically protected (e.g. using TLS {{!DoT=RFC7858}} or HTTPS
-{{!DoH=RFC8484}}), the client SHOULD NOT fall back to $ADDR_QNAME (step 5
-above) or the authority endpoint.  Otherwise, an active attacker
+{{!DoH=RFC8484}}), a SVCB-optional client SHOULD terminate the connection like
+a SVCB-reliant client would.  Otherwise, an active attacker
```

Comment:
terminate the connection or terminate the connection attempt?

Reply:
Connection attempt.  Reworded.